rs/ic_os/metrics_tool/src/lib.rs

```diff
@@ -0,0 +1,160 @@
+// TODO: refactor/merge this with fstrim_tool and guestos_tool metrics functionality
+use std::fs::File;
+use std::io::{self, Write};
+use std::path::Path;
+
+// TODO: everything is floating point for now
+pub struct Metric {
+    name: String,
+    value: f64,
+    annotation: String,
+    labels: Vec<(String, String)>,
+}
+
+impl Metric {
+    pub fn new(name: &str, value: f64) -> Self {
+        Self {
+            name: name.to_string(),
+            value,
+            annotation: "Custom metric".to_string(),
+            labels: Vec::new(),
+        }
+    }
+    pub fn with_annotation(name: &str, value: f64, annotation: &str) -> Self {
+        Self {
+            name: name.to_string(),
+            value,
+            annotation: annotation.to_string(),
+            labels: Vec::new(),
+        }
+    }
+
+    pub fn add_annotation(mut self, annotation: &str) -> Self {
+        self.annotation = annotation.to_string();
+        self
+    }
+
+    pub fn add_label(mut self, key: &str, value: &str) -> Self {
+        self.labels.push((key.to_string(), value.to_string()));
+        self
+    }
+
+    // TODO: formatting of floats
+    pub fn to_string(&self) -> String {
```

Comment:
Could you name it something other than to_string?
to_string is a fairly standard method and is automatically provided for types that implement Display. This type does not implement Display yet but if it does in the future, the two to_string implementations will be confusing. Also, to_string and Display are for printing something on the screen/a file (eg. for users), whereas this is Prometheus specific which would be nice to see from the name.

Reply:
Agreed, renamed to `to_prom_string` and commented accordingly